tests/testthat/test-run.R

```diff
@@ -63,6 +63,7 @@ test_that("run", {
   expect_identical(unname(unlist(run$hash_artefacts, use.names = FALSE)),
                    hash_files(file.path(p, "mygraph.png"), FALSE))
   expect_identical(run$hash_resources, list())
+  expect_identical(run$size_resources, list())
```

Comment:
these will come out.

What needs to come in is a test where you create a script file that when run does modify the inputs (one test modifying a resource, then in another test modify a global resource) and check that your condition does trigger as expected